python_modules/dagster/dagster/grpc/impl.py

```diff
@@ -115,18 +116,22 @@ def _run_in_subprocess(
         execute_run_args = deserialize_json_to_dagster_namedtuple(serialized_execute_run_args)
         check.inst_param(execute_run_args, "execute_run_args", ExecuteExternalPipelineArgs)
 
-        instance = DagsterInstance.from_ref(execute_run_args.instance_ref)
-        pipeline_run = instance.get_run_by_id(execute_run_args.pipeline_run_id)
-
-        if not pipeline_run:
-            raise DagsterRunNotFoundError(
-                "gRPC server could not load run {run_id} in order to execute it. Make sure that the gRPC server has access to your run storage.".format(
-                    run_id=execute_run_args.pipeline_run_id
-                ),
-                invalid_run_id=execute_run_args.pipeline_run_id,
-            )
+        with (
+            DagsterInstance.from_ref(execute_run_args.instance_ref)
+            if execute_run_args.instance_ref
+            else nullcontext()
+        ) as instance:
+            pipeline_run = instance.get_run_by_id(execute_run_args.pipeline_run_id)
+
+            if not pipeline_run:
+                raise DagsterRunNotFoundError(
+                    "gRPC server could not load run {run_id} in order to execute it. Make sure that the gRPC server has access to your run storage.".format(
+                        run_id=execute_run_args.pipeline_run_id
+                    ),
+                    invalid_run_id=execute_run_args.pipeline_run_id,
+                )
 
-        pid = os.getpid()
+            pid = os.getpid()
 
```

Comment:
now that instance is in a contextmanager, you should remove 146-147? bit surprised that didn't throw a lint or something, but you definitely don't want to dispose the instance twice